drivers/gpio/CMakeLists.txt

```diff
@@ -27,6 +27,7 @@ zephyr_library_sources_ifdef(CONFIG_GPIO_SAM4L      gpio_sam4l.c)
 zephyr_library_sources_ifdef(CONFIG_GPIO_SX1509B    gpio_sx1509b.c)
 zephyr_library_sources_ifdef(CONFIG_GPIO_INTEL      gpio_intel.c)
 zephyr_library_sources_ifdef(CONFIG_GPIO_STELLARIS  gpio_stellaris.c)
+zephyr_library_sources_ifdef(CONFIG_GPIO_RASPBERRYPI gpio_raspberrypi.c)
```

Comment:
I have limited in-depth knowledge of the Raspberry Pi Pico SDK, and I'm afraid have absolutely zero knowledge of Zephyr, but I wonder if calling this `CONFIG_GPIO_RASPBERRYPI` and implementing it in a `gpio_raspberrypi.c` might cause confusion with the "big" Raspberry Pi boards, based on the BCM283x / BCM2711 SoCs? :man_shrugging: (as they would need a different GPIO driver to that required for RP2040)

Reply:
AFAIK there is no Zephyr port for the big boards, so it's not a problem, at least for now

Reply:
The `Raspberry PI` is the name of a family of boards so it cannot be used as is. Please, add the `pico` name or something equivalent so that it can be different from the other RPi boards.

Reply:
Since this is applicable to the rp2040 chip, not just the pico board itself, I would argue against adding pico in the naming here.

There are *hint* that other combinations/chips might come in the same line, with the "2040" portion noting the number of cores, etc.

Not much left to differentiate "rapsberry pi, the SBC that runs a Broadcom chip", and "rapsberry pi, the SoC".

Reply:
ok, not familiar with this SoC. Referring to it as `rp2040` would work?

Reply:
Dug a bit in the SDK, which certainly is in fact named "pico-sdk", as are many if the libraries there. Yet at the same time, the rp2040 datasheet always simply refers to it as the rp2040, yet has a USB interface PICO-BOOTROM.

In the interest of recognition, I'd retract my objection, and say "pico probably better than anything else".

Reply:
I would also vote for "pico" as the drivers wrap pico-sdk, so `CONFIG_GPIO_RASPBERRYPI_PICO` really means we're using the pico driver.
From their website it really seems that Pico was supposed to be the name of the board, with no particular name for the family of chips. But since that name is already in their code base, it might eventually become the name of the family. When new chips are released, we'll probably know, and if the name of the sdk changes, we might have to change this as well.